fs/move.py

```diff
@@ -55,24 +59,50 @@ def move_file(
             resources (defaults to `False`).
 
     """
-    with manage_fs(src_fs) as _src_fs:
-        with manage_fs(dst_fs, create=True) as _dst_fs:
+    with manage_fs(src_fs, writeable=True) as _src_fs:
+        with manage_fs(dst_fs, writeable=True, create=True) as _dst_fs:
             if _src_fs is _dst_fs:
                 # Same filesystem, may be optimized
                 _src_fs.move(
                     src_path, dst_path, overwrite=True, preserve_time=preserve_time
                 )
-            else:
-                # Standard copy and delete
-                with _src_fs.lock(), _dst_fs.lock():
-                    copy_file(
-                        _src_fs,
-                        src_path,
-                        _dst_fs,
-                        dst_path,
-                        preserve_time=preserve_time,
-                    )
+                return
+
+            if _src_fs.hassyspath(src_path) and _dst_fs.hassyspath(dst_path):
+                # if both filesystems have a syspath we create a new OSFS from a
+                # common parent folder and use it to move the file.
+                try:
+                    src_syspath = _src_fs.getsyspath(src_path)
+                    dst_syspath = _dst_fs.getsyspath(dst_path)
+                    common = commonpath([src_syspath, dst_syspath])
+                    rel_src = frombase(common, src_syspath)
+                    rel_dst = frombase(common, dst_syspath)
+                    with _src_fs.lock(), _dst_fs.lock(), open_fs(
+                        common, writeable=True
+                    ) as base:
+                        base.move(rel_src, rel_dst, preserve_time=preserve_time)
+                    return  # optimization worked, exit early
```

Comment:
Perhaps it makes sense (stylistically) to move this to the same level of indentation as `base.move` ?

Reply:
Beautified this a bit ;)

Reply:
Nice, `with a(), b(), c() as d:` _was_ a bit confusing to read :wink: 
I still think that the `return` ought to be at the same indent-level as the `base.move` call though?

Reply:
Sure, why not - comes in the next commit